frontend/javascripts/admin/dataset/dataset_components.js

```diff
@@ -82,7 +82,7 @@ export function DatastoreFormItem({
       <Select
         showSearch
         placeholder="Select a Datastore"
-        optionFilterProp="children"
+        optionFilterProp="label"
```

Comment:
according to the antd docs, `label` should be passed here if the `options` prop is used. this also explains the breakage since we used to pass JSX children to the `select` components prior to the antd upgrade.